components/encryption/src/manager/mod.rs

```diff
@@ -0,0 +1,594 @@
+use std::collections::hash_map::Entry;
+use std::io::{Error as IoError, ErrorKind, Result as IoResult};
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, RwLock};
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+use kvproto::encryptionpb::{DataKey, EncryptionMethod, FileDictionary, FileInfo, KeyDictionary};
+use protobuf::Message;
+use rocksdb::{EncryptionKeyManager, FileEncryptionInfo};
+
+use crate::crypter::*;
+use crate::encrypted_file::EncryptedFile;
+use crate::master_key::*;
+use crate::{Error, Iv, Result};
+
+const KEY_DICT_NAME: &str = "key.dict";
+const FILE_DICT_NAME: &str = "file.dict";
+
+struct Dicts {
+    file_dict: FileDictionary,
+    key_dict: KeyDictionary,
+    rotation_period: Duration,
+    base: PathBuf,
+}
+
+impl Dicts {
+    fn open(path: &str, rotation_period: Duration, master_key: &dyn Backend) -> Result<Dicts> {
+        let base = Path::new(path);
+        let file_file = EncryptedFile::new(base, FILE_DICT_NAME);
+        let file_bytes = file_file.read(master_key)?;
```

Comment:
here master key is used to read file_dict, and in `save_file_dict` plaintext is used. Need to be consistent.